src/main/java/org/commcare/formplayer/services/MediaHandler.kt

```diff
@@ -15,28 +18,29 @@ class MediaHandler(val file: MultipartFile) {
     companion object {
         // 3 MB size limit
         const val MAX_BYTES = (3 * 1048576 - 1024).toLong()
-        private val SUPPORTED_FILE_EXTS = arrayOf(".jpg", "jpeg", "png")
+        private val SUPPORTED_FILE_EXTS = ImmutableList.of(".jpg", "jpeg", "png", "pdf")
+        private val SUPPORTED_MIME_TYPES = ImmutableList.of("image", "application/pdf")
     }
 
     /**
      * Saves file in the given parent directory
      */
-    fun saveFile(parentDir: String): String {
+    fun saveFile(parentDirPath: Path): String {
         validateFile()
         val fileId = PropertyUtils.genUUID()
-        val parent = File(parentDir)
+        val parent = parentDirPath.toFile()
         parent.mkdirs()
-        val mediaPath: String = getMediaFilePath(parentDir, fileId)
+        val desintationFile = getMediaFilePath(parentDirPath, fileId).toFile()
         try {
-            FileUtils.copyFile(file.inputStream, File(mediaPath))
+            FileUtils.copyFile(file.inputStream, desintationFile)
             return fileId
         } catch (e: IOException) {
             throw IOException("Could not copy file to destination due to " + e.message, e)
         }
     }
 
     private fun validateFile() {
-        if (isUnSupportedFileExtension()) {
+        if (isUnSupportedFileExtension() && isUnsupportedMimeType()) {
```

Comment:
should this be an `or`

Reply:
I mostly tried to replicate what mobile is doing [today](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/utils/FormUploadUtil.java#L423). Seems like we would be ok as long as the mime type is correct even if the extension is not in our list.